src/index.js

```diff
@@ -1,106 +1,297 @@
-        });
-        return await Promise.all(maps);
-    } catch (err) {
-        throw new Error(
-            `Unable to load import map file from server: ${err.message}`,
-        );
-    }
-}
+	try {
+		const maps = urls.map(async (map) => {
+			const { statusCode, body } = await request(map, {
+				maxRedirections: 2,
+			});
+
+			if (statusCode === 404) {
+				throw new Error("Import map could not be found on server");
+			} else if (statusCode >= 400 && statusCode < 500) {
+				throw new Error("Server rejected client request");
+			} else if (statusCode >= 500) {
+				throw new Error("Server error");
+			}
+			return body.json();
+		});
+		return await Promise.all(maps);
+	} catch (err) {
+		throw new Error(
+			`Unable to load import map file from server: ${err.message}`,
+		);
+	}
+};
+
+/**
+ * @typedef {object} Options
+ * @property {string} [base=null]
+ * @property {boolean} [development=false]
+ * @property {boolean} [loadMaps=false]
+ * @property {string} [path=process.cwd()]
+ */
+
+/**
+ * @typedef {object} ImportMap
+ * @property {Record<string, string>} imports
+ */
+
+/**
+ * An Eik utility for servers running on Node. With it you can:
+ *  - generate different URLs to assets on an Eik server depending on environment (development vs production).
+ *  - get the import maps you have configured in `eik.json` from the Eik server, should you want to use them in the HTML response.
+ *
+ * @example
+ * ```js
+ * // Create an instance, then load information from `eik.json` and the Eik server
+ * import Eik from "@eik/node-client";
+ *
+ * const eik = new Eik();
+ * await eik.load();
+ * ```
+ * @example
+ * ```js
+ * // Serve a local version of a file from `./public`
+ * // in development and from Eik in production
+ * import path from "node:path";
+ * import Eik from "@eik/node-client";
+ * import fastifyStatic from "@fastify/static";
+ * import fastify from "fastify";
+ *
+ * const app = fastify();
+ * app.register(fastifyStatic, {
+ *    root: path.join(process.cwd(), "public"),
+ *    prefix: "/public/",
+ * });
+ *
+ * const eik = new Eik({
+ *    development: process.env.NODE_ENV === "development",
+ *    base: "/public",
+ * });
+ *
+ * // load information from `eik.json` and the Eik server
+ * await eik.load();
+ *
+ * // when development is true script.value will be /public/script.js.
+ * // when development is false script.value will be
+ * // https://{server}/pkg/{name}/{version}/script.js
+ * // where {server}, {name} and {version} are read from eik.json
+ * const script = eik.file("/script.js");
+ *
+ * app.get("/", (req, reply) => {
+ *    reply.type("text/html; charset=utf-8");
+ *    reply.send(`<html><body>
+ *      <script
+ *        src="${script.value}"
+ *        ${script.integrity ? `integrity="${script.integrity}"` : ""}
+ *        type="module"></script>
+ *    </body></html>`);
+ * });
+ *
+ * app.listen({
+ *    port: 3000,
+ * });
+ *
+ * console.log("Listening on http://localhost:3000");
```

Comment:
I was curious as to why repeat the readme docs here, as I would think one place would suffice and be easier to keep up to date over time 🤔

Reply:
Certainly easier to update, but I think there's value in having the docs both show up in an editor and available on the web. Honestly I'm thinking about adding a third location – the docs on https://eik.dev 😅 (For MetricsJS I ended up [moving much of the README.md to be only on the docsite](https://github.com/metrics-js/client#readme). Maybe we do something like that?)